src/sources/BatchCreator.cpp

```diff
@@ -1,44 +1,84 @@
 /**
  * @file BatchCreator.hpp
- * @author
- * @date
+ * @author Maximilian Rodler
+ * @date 22.04.2024
  * @version
- * @brief
- * @details
+ * @brief Creates batch file
+ * @details Creates batch file from Arguments in JSON
  *
  * @copyright See LICENSE file
  */
 
 #include "BatchCreator.hpp"
+#include "LoggingWrapper.hpp"
 
-BatchCreator::BatchCreator(std::shared_ptr<parsing::FileData> fileData) {
+BatchCreator::BatchCreator(std::shared_ptr<parsing::FileData> fileData){
 
 }
 
+void BatchCreator::createBatch() {
+    LOG_INFO << "Creating Batch file";
 
-
-void BatchCreator::writeBatch() {
+    this->batchFile.open(this->fileData->getOutputFile());
     this->writeStart();
     this->writeHideShell();
 
 }
 
 void BatchCreator::writeStart() {
-    this->batchFileString = this->batchFileString + "@ECHO OFF\nC:\\Windows\\System32\\cmd.exe ";
+    LOG_INFO << "writing Start of Batch";
+    this->batchFile << "@ECHO OFF\nC:\\Windows\\System32\\cmd.exe ";
 }
-
-void BatchCreator::writeHideShell() {
-    if (this->fileData->getHideShell()) {
-        this->batchFileString = this->batchFileString + "/c\n";
+ 
+ void BatchCreator::writeHideShell() {
+    if (this->fileData->getHideShell()){
+        LOG_INFO << "writing hide Shell";
+        this->batchFile << "/c \n";
 
     } else {
-        this-> batchFileString = this->batchFileString + "/k\n";
+        LOG_INFO << "writing show Shell";
+        this->batchFile << "/k \n";
     }
 }
 
-void BatchCreator::writeCommand() {
-    this->batchFileString = this->batchFileString + "\"";
+void BatchCreator::writeCommands() {
+    LOG_INFO << "writing Commands";
+    this->batchFile << "\"";
     for (const std::string command : this->fileData->getCommands()) {
-        this->batchFileString = this->batchFileString + command + "&& \n";
+        this->batchFile << command << "&& \n";
+    }
+}
+
+void BatchCreator::writeEnvVariables() {
+    LOG_INFO << "writing Environment Variables";
+    for (const std::tuple env : this->fileData->getEnvironmentVariables()) {
+        this->batchFile << "set " << std::get<0>(env) << "=" 
+                        << std::get<1>(env) << " && \n";
+    }
+}
+
+void BatchCreator::writePathVariables() {
+    LOG_INFO << "writing Path Variables";
+    this->batchFile << "set path=";
+    for (const std::string path : this->fileData->getPathValues()) {
+        this->batchFile << path << ";\n";
     }
+    this->batchFile << "%path%";
+}
+
+void BatchCreator::writeApp() {
+    std::string appName = this->fileData->getOutputFile();
```

Comment:
what does this do?

Reply:
Holt den Namen der Anwendung damit dieser beim Staten mit ausgegeben werden kann.
(Wird in der nächsten Zeile formatiert: `appName = appName.substr(...)` )